app/src/main/java/org/hwyl/sexytopo/control/util/Space2DUtils.java

```diff
@@ -94,4 +96,57 @@ public static double adjustAngle(double angle, double delta) {
         Coord2D end = line.getEnd();
         return new Line<>(start.plus(point), end.plus(point));
     }
+
+    private static List<Coord2D> DouglasPeukerIteration(List<Coord2D> path, double epsilon) {
+
+        // Find the point with the maximum distance
+        int pathSize = path.size();
+        int indexMax = 0;
+        double distMax = 0;
+
+        for (int i = 1; i != pathSize; ++i) {
+            double dist = getDistanceFromLine(path.get(i), path.get(0), path.get(pathSize - 1));
```

Comment:
Considering how much this function gets called we should really implement a version that does not perform the square root and compare against the square of the epsilon but we can optimise when and where necessary